src/zenml/integrations/whylogs/whylogs_utils.py

```diff
@@ -0,0 +1,120 @@
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import datetime
+from typing import Dict, Optional
+
+import pandas as pd
+from whylogs import DatasetProfile  # type: ignore
+from whylogs.app import Session  # type: ignore
+
+from zenml.steps.step_context import StepContext
+
+
+class WhylogsContext(StepContext):
+    """Extends the base step context with whylogs session and logger management
+    to facilitate whylogs data logging and profiling inside a step function.
+
+    To use it, add a `WhylogsContext` object to the signature of your step
+    function like this:
+
+    @step
+    def my_step(context: WhylogsContext, ...)
+        ...
+        context.log_dataframe(...)
+
+    See `StepContext` for additional information on how step contexts can
+    be used.
+
+    **Note**: When using a `WhylogsContext` inside a step, ZenML disables
+    caching for this step by default as the context provides access to external
+    resources which might influence the result of your step execution. To
+    enable caching anyway, explicitly enable it in the `@step` decorator or when
+    initializing your custom step class.
+    """
+
+    _session: Session = None
+
+    def get_whylogs_session(
+        self,
+    ) -> Session:
+        """Returns the whylogs session associated with the current step.
+
+        Args:
+            output_name: Optional name of the output for which to get the
+                materializer. If no name is given and the step only has a
+                single output, the materializer of this output will be
+                returned. If the step has multiple outputs, an exception
+                will be raised.
+            custom_materializer_class: If given, this `BaseMaterializer`
+                subclass will be initialized with the output artifact instead
+                of the materializer that was registered for this step output.
+
+        Returns:
+            A materializer initialized with the output artifact for
+            the given output.
+        """
+        if self._session is not None:
+            return self._session
+
+        self._session = Session(
+            project=self.step_name,
+            # TODO: replace with pipeline name when available in the context
+            pipeline=self.step_name,
+            # keeping the writers list empty, serialization is done in the
+            # materializer
+            writers=[],
+        )
+
+        return self._session
+
+    def log_dataframe(
```

Comment:
If I understood correctly then this method does not actually log/write/upload anything unless it is materialized, correct? If that's the case I feel like we should clarify that by either changing the method name or mentioning it in the docstring somewhere.

Reply:
Correct ! And a very good point.